content/shmem_bundle_start.tex

```diff
@@ -0,0 +1,35 @@
+\apisummary{
+    Start a communication bundle.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void @\FuncDecl{shmem\_bundle\_start}@(void);
+void @\FuncDecl{shmem\_ctx\_bundle\_start}@(shmem_ctx_t ctx);
```

Comment:
I think - we are missing two things:
1. There should a hint parameter for this routine - apart from the context. Hints should help implementations to support different options.
2. Should there be shmem_team_t as part of the argument? I'm not entirely sure about this requirement. **My initial thinking is we should support - I'm open to suggestions.** Adding teams argument - doesn't make it a collective call. Though, we can also retrieve this information from the ctx argument - it is useful to have teams argument along with or instead of teams. Proposed API:
```C++
vous shmem_bundle_start(shmem_team_t team, shmem_ctx_t ctx, int hints)
```
Hint can be used to determine whether team or ctx is passed. If multiple CTX or all CTX from a team is expected to create separate bundles - it would be optimal to use a single internal bundle.

Reply:
I'm confused on this one - doesn't the context directly imply a team?  I don't see why we would allow bundling on a team other than the context it's associated with...?

Reply:
Nevertheless - we need hints argument irrespective of whether we plan to add team argument or not.

Reply:
hint would say - SHMEM_BUNDLE_TEAM or SHMEM_BUNDLE_CTX

Reply:
I think we should have hints inside this routine and not in the context - I don't think we want to restrict the users to use a separate context for these sort of optimizations. They should be able to use the same context within the session and outside the session

Reply:
If they misuse them - say use a shared context within session and simulateously outside on another thread - it is their perf issue to handle.

Reply:
I agree. Sorry, my comment was a bit confusing. I was suggesting that the parameter to the proposed function could be called `long options` if we want to follow the convention set by the contexts API.